examples/python/trigger-pagerduty-incident/handler/handler.py

```diff
@@ -0,0 +1,179 @@
+        """
+        Make a rest api call to Pagerduty Events API
+        
+        Arguments:
+            obj {dict} -- Generated API Body for the PagerDuty Events API
+
+        Returns:
+            FaaSResponse -- status code and message
+        """
+        obj['routing_key'] = self.routingkey
+        obj['event_action'] = self.event_action
+        try:
+            resp = self.session.post(PAGERDUTY_API_PATH,json=obj)
+            resp.raise_for_status()
+            print(resp.text)
+            resp_body = json.loads(resp.text)
+            return FaaSResponse('200', 'Successfully {0}ed event action with dedup_key: {1}'.format(self.event_action, resp_body['dedup_key']))
+        except requests.HTTPError as err:
+            return FaaSResponse('500', 'Could not invoke Pagerduty API > HTTPError: {0}'.format(err))
+
+def handle(req):
+    # Validate Event input
+    try:
+        cevent = json.loads(req)
+    except ValueError as err:
+        res = FaaSResponse('400','invalid JSON > ValueError: {0}'.format(err))
+        print(json.dumps(vars(res)))
+        return
+    
+    # Assert that the Event conforms to CloudEvent Spec
+    # For debugging: validate the JSON blob we received - uncomment print statements if needed
+    # print(cevent)
+    try:
+        eventdata = (cevent['data'])
+        # Map the CloudEvent data and build the PagerDuty Event API Request body
+        obj = {
+                'routing_key': 'REPLACE',
+                'event_action': 'REPLACE',
+                'client': 'VMware Event Broker Appliance',
+                'client_url': cevent['source'],
+                'payload': {
+                    'summary': eventdata['FullFormattedMessage'],
+                    'timestamp': eventdata['CreatedTime'],
+                    'source': cevent['source'],
+                    'severity': 'info',
+                    'component': eventdata['Vm']['Name'],
+                    'group': eventdata['Host']['Name'],
+                    'class': cevent['subject'],
+                    'custom_details': {
+                        'user': eventdata['UserName'],
+                        'Datacenter': eventdata['Datacenter'],
+                        'ComputeResource': eventdata['ComputeResource'],
+                        'Host': eventdata['Host'],
+                        'VM': eventdata['Vm']
+                    }
+                }
+                #,
+                #'images': [{
+                #    'src': 'https://www.pagerduty.com/wp-content/uploads/2016/05/pagerduty-logo-green.png',
+                #    'href': 'https://example.com/',
+                #    'alt': eventdata['Vm']
+                # }],
+                # 'links': [{
+                #    'href': 'https://example.com/',
+                #    'text': 'Link to VM'
+                # }]
+            }
+    except KeyError as err:
+        res = FaaSResponse('400','JSON does not conform to CloudEvent spec > KeyError: {0}'.format(err))
```

Comment:
This error message is strictly speaking not correct. If you fail to look up a key, e.g. `eventdata['Host']['Name']` in an event which has this set to`NULL`, the event still conforms to the CloudEvent spec. I think it's better to change the response message to something like "required key not found in JSON...".

Reply:
Updated the exception handling to be more accurate and avoided nested/complex flows